doc/developers/contributing.rst

```diff
@@ -125,16 +110,33 @@ New Contributors
 ----------------
 
 We recommend new contributors start by reading this contributing guide, in
-particular :ref:`ways_to_contribute`, :ref:`automated_contributions_policy`
-and :ref:`pr_checklist`. For expected etiquette around which issues and stalled PRs
+particular :ref:`ways_to_contribute`, :ref:`automated_contributions_policy`.
+
+Next, we advise new contributors gain foundational knowledge on
+scikit-learn and open source by making
+:ref:`non-code contributions <ways_to_contribute>`:
+
+* :ref:`improving and investigating issues <bug_triaging>`
+
+  * confirming that a problem reported can be reproduced and providing a
+    :ref:`minimal reproducible code <minimal_reproducer>` (if missing), can help you
+    learn about different use cases and user needs
+  * investigating the root cause of an issue will aid you in familiarising yourself
+    with the scikit-learn codebase
+
+* :ref:`reviewing other developers' pull requests <code_review>` will help you
+  develop an understanding of the requirements and quality expected of contributions
+* improving the :ref:`contribute_documentation` can help deepen your knowledge
+  of the statistical concepts behind models and functions, and scikit-learn API
+
+If you wish to make code contributions after building your foundational knowledge, we
+recommend you start by looking for an issue that is of interest to you, in an area you
+are already familiar with as a user or have background knowledge of. We recommend
+starting with smaller pull requests and following our :ref:`pr_checklist`.
+For expected etiquette around which issues and stalled PRs
 to work on, please read :ref:`stalled_pull_request`, :ref:`stalled_unclaimed_issues`
 and :ref:`issues_tagged_needs_triage`.
```

Comment:
Having a re-read, I moved the 'to read' contributing sections that pertain to contributing code down here, as we don't talk about code contributions until here.